File: sandbox-runner/patterns/provider-fallback/go/worker.go
package main

import (
	"log"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	// One registry shared by every Activity this worker runs, injected into the
	// Activities value instead of reached for as a package global. See the note in
	// activities.go on why this is safe only for same-worker, per-Workflow-keyed state.
	activities := &Activities{registry: NewLLMRegistry()}

	w := worker.New(c, TaskQueue, worker.Options{})
	w.RegisterWorkflow(ProviderFallbackWorkflow)
	w.RegisterActivity(activities)

	log.Printf("Worker listening on task queue '%s'", TaskQueue)
	if err := w.Run(worker.InterruptCh()); err != nil {
		log.Fatalln("Worker run failed:", err)
	}
}

File: sandbox-runner/patterns/provider-fallback/python/worker.py
import asyncio

from temporalio.client import Client
from temporalio.worker import Worker

from activities import CompletionActivities, LLMRegistry
from shared import TASK_QUEUE
from workflows import ProviderFallbackWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")

    # One registry shared by every Activity this worker runs, injected into the
    # activities instance instead of reached for as a module global. See the note in
    # activities.py on why this is safe only for same-worker, per-Workflow-keyed state.
    registry = LLMRegistry()
    activities = CompletionActivities(registry)

    worker = Worker(
        client,
        task_queue=TASK_QUEUE,
        workflows=[ProviderFallbackWorkflow],
        activities=[activities.generate, activities.run_tool],
    )
    print(f"Worker listening on task queue '{TASK_QUEUE}'", flush=True)
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())
